feature-libs/product-configurator/rulebased/components/attribute/types/single-selection-bundle-dropdown/configurator-attribute-single-selection-bundle-dropdown.component.ts

```diff
@@ -50,80 +33,17 @@ export class ConfiguratorAttributeSingleSelectionBundleDropdownComponent
     }
   }
 
-  get withQuantity() {
-    return this.quantityService.withQuantity(
-      this.attribute?.dataType ?? Configurator.DataType.NOT_IMPLEMENTED,
-      this.attribute?.uiType ?? Configurator.UiType.NOT_IMPLEMENTED
-    );
-  }
-
-  get disableQuantityActions() {
-    return this.quantityService.disableQuantityActions(
-      this.attributeDropDownForm?.value
-    );
-  }
-
-  onSelect(): void {
-    const event: ConfigFormUpdateEvent = {
-      changedAttribute: {
-        ...this.attribute,
-        selectedSingleValue: this.attributeDropDownForm?.value,
-      },
-      ownerKey: this.ownerKey,
-      updateType: Configurator.UpdateType.ATTRIBUTE,
-    };
-    this.selectionChange.emit(event);
-  }
-
-  onDeselect(): void {
-    const event: ConfigFormUpdateEvent = {
-      changedAttribute: {
-        ...this.attribute,
-        selectedSingleValue: '0',
-      },
-      ownerKey: this.ownerKey,
-      updateType: Configurator.UpdateType.ATTRIBUTE,
-    };
-
-    this.selectionChange.emit(event);
-  }
-
-  onChangeQuantity(eventObject: any): void {
-    this.loading$.next(true);
-
-    if (!eventObject) {
-      this.attributeDropDownForm.setValue('');
-      this.onSelect();
-    } else {
-      const event: ConfigFormUpdateEvent = {
-        changedAttribute: {
-          ...this.attribute,
-          quantity: eventObject,
-        },
-        ownerKey: this.ownerKey,
-        updateType: Configurator.UpdateType.ATTRIBUTE_QUANTITY,
-      };
-
-      this.selectionChange.emit(event);
-    }
-  }
-
-  getSelectedValuePrice(): Configurator.PriceDetails | undefined {
-    return this.attribute?.values?.find((value) => value?.selected)?.valuePrice;
-  }
-
   /**
-   * Extract corresponding price formula parameters
-   *
-   * @return {ConfiguratorPriceComponentOptions} - New price formula
+   TODO(issue: #11238): update @deprecated level to the release we are publishing with,
+   It is still 3.1 only because app.module.ts states that we are on 3.1.
+   Finally we must have 3.x, x>=2 here
    */
-  extractPriceFormulaParameters(): ConfiguratorPriceComponentOptions {
-    return {
-      quantity: this.attribute?.quantity,
-      price: this.getSelectedValuePrice(),
-      priceTotal: this.attribute?.attributePriceTotal,
-      isLightedUp: true,
-    };
+  /**
+   * @deprecated since 3.1
+   * User better onSelect('0')
+   */
+  onDeselect(): void {
```

Comment:
if we deprecate this `onDeslect`, then we will not use it any more. So, in html, we should uses `onSelect('0')`.

Reply:
Done!